aggregator_core/src/datastore.rs

```diff
@@ -927,47 +904,33 @@ impl<C: Clock> Transaction<'_, C> {
             )
             .map(SecretBytes::new)?;
 
-        // Aggregator authentication tokens.
-        let mut aggregator_auth_tokens = Vec::new();
-        for row in aggregator_auth_token_rows {
-            let ord: i64 = row.get("ord");
-            let auth_token_type: AuthenticationTokenType = row.get("type");
-            let encrypted_aggregator_auth_token: Vec<u8> = row.get("token");
+        let aggregator_auth_token = if let Some(row) = aggregator_auth_token_row {
```

Comment:
(assuming the error types work out, it's hard to reason about the operation of the `?` operator without an IDE)

Reply:
I was a little surprised to see that I could write an `Option::map` closure with two different fallible calls in it by using `Option::transpose`, but it works, along with an `Option::zip` since we no longer have a distinct `Option<Row>` for the auth tokens.